src/java.base/share/classes/java/util/zip/Deflater.java

```diff
@@ -868,23 +870,44 @@ public void reset() {
     }
 
     /**
-     * Closes the compressor and discards any unprocessed input.
+     * Closes and releases the resources held by this {@code Deflater}
+     * and discards any unprocessed input.
+     * <p>
+     * If this method is invoked multiple times, the subsequent calls are treated as a no-op.
+     * Several other methods defined by this class will throw an {@link IllegalStateException}
+     * if invoked on a closed {@code Deflater}.
```

Comment:
All of the methods which call ensureOpen should either document that an IllegalStateException may be thrown or you could add it to the header of the class.

Reply:
Hello Lance, you are right - I forgot to do that when I changed it to throw IllegalStateException from the previous NullPointerException. I've now updated the PR to document this on individual methods in the Inflater/Deflater which throw this exception.

Reply:
Thank you.  It was an oversight that the original javadoc did not call out the NPE, but now is our chance to get it right with a more reasonable exception.

Reply:
Overall I agree with defining that IllegalStateException is thrown after the Deflater is closed, and adding documentation of this to all the methods that throw it.

Reply:
I am no sure the last sentence, “several…” is needed here given we now document the IAE is thrown